telegram/ext/conversationhandler.py

```diff
@@ -650,11 +650,16 @@ def handle_update(  # type: ignore[override]
                         "Ignoring `conversation_timeout` because the Dispatcher has no JobQueue."
                     )
 
-        if isinstance(self.map_to_parent, dict) and new_state in self.map_to_parent:
+        if isinstance(new_state, Promise):
+            new_state_str = self._resolve_promise((new_state, new_state))
```

Comment:
if the promise is not done yet, this doesn't change anything about the current behavior.

btw, the naming `new_state_str` is not very suitable because states can be any hashable object ;)